packages/typescript-eslint/src/index.ts

```diff
@@ -71,7 +71,6 @@ const configs = {
 export type Config = TSESLint.FlatConfig.ConfigFile;
 export type { ConfigWithExtends };
 /*
-eslint-disable-next-line import/no-default-export --
```

Comment:
Looks like this rule is actually not working correctly in any file — `context.parserOptions.sourceType` always comes back as `undefined`, whereas this rule needs it to be `module`

Reply:
That's because we don't set it via parser options.
https://github.com/typescript-eslint/typescript-eslint/blob/main/eslint.config.mjs#L90-L94

It shouldn't be reading the parser options and should instead be reading it from the parsed result, really.

Probs worth reporting upstream.